orangewidget/utils/messagewidget.py

```diff
@@ -547,54 +630,20 @@ def flashIcon(self):
                 self.anim.start(QPropertyAnimation.KeepWhenStopped)
                 break
 
-    @staticmethod
-    def __styled(css, html):
-        # Prepend css style sheet before a html fragment.
-        if css.strip():
-            return "<style>\n" + escape(css) + "\n</style>\n" + html
-        else:
-            return html
-
     def __update(self):
-        """
-        Update the current display state.
-        """
-        self.ensurePolished()
-        summary = self.summarize()
-        icon = message_icon(summary)
-        self.__iconwidget.setIcon(icon)
-        self.__iconwidget.setVisible(not (summary.isEmpty() or icon.isNull()))
-        self.__textlabel.setTextFormat(summary.textFormat)
-        self.__textlabel.setText(summary.text)
-        self.__textlabel.setVisible(bool(summary.text))
-
-        def is_short(m):
-            return not (m.informativeText or m.detailedText)
-
-        messages = [m for m in self.__messages.values() if not m.isEmpty()]
-        if not messages:
-            fulltext = ""
-        else:
-            messages = sorted(messages, key=attrgetter("severity"),
-                              reverse=True)
-            fulltext = "<hr/>".join(m.asHtml() for m in messages)
-        self.__fulltext = fulltext
-        self.setToolTip(self.__styled(self.__defaultStyleSheet, fulltext))
-        self.anim.start(QPropertyAnimation.KeepWhenStopped)
-
-        self.layout().activate()
+        super().setMessage(self.summarize())
 
     def mousePressEvent(self, event):
         if event.button() == Qt.LeftButton:
-            if self.__fulltext:
+            message = self.message
+            if message and (message.detailedText or message.informativeText):
```

Comment:
```suggestion
            if message:
```
The popup should always display if there are any contents. The inline display can frequently be too short to display the full contents.

Reply:
I thought a tooltip would suffice, but I agree there's not need to disable the popup. I changed the condition as suggested.